components/Carousel/Carousel.js

```diff
@@ -1,11 +1,42 @@
 class Carousel {
+    constructor(carouselElement){
+        this.carouselElement = carouselElement
+        this.navLeft = this.carouselElement.querySelector('.left-button');
+        this.navRight = this.carouselElement.querySelector('.right-button');
+        this.images = this.carouselElement.querySelectorAll('img');
+
+        this.navLeft.addEventListener('click', (e) => this.navigate(-1));
+        this.navRight.addEventListener('click', (e) => this.navigate(1));
+        this.navigate(0);
+      
+    }
+    navigate(direction){
+        let counter = 0;
+        let carouselLength = this.carouselElement.length;
+        let currentImg = this.images[0];
```

Comment:
You need to track this `currentImg` variable in your constructor rather than in a method. With current setup, you run into problems with even trying to switch between images!